src/App.tsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = items.length;
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [curPage, setCurPage] = useState(1);
+
+  const numOfPages = Math.ceil(total / itemsPerPage);
+
+  const elements = curPage * itemsPerPage;
+
+  const fromItem = (curPage - 1) * itemsPerPage + 1;
+  const toItem = elements > total ? total : elements;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {curPage} (items {fromItem} - {toItem} of {total})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            onChange={event => {
```

Comment:
Pls move the function inside `onChange` handler to a separate const and use as a callback here
*PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**